### PR TITLE
buildroot: Speed up build times

### PR DESCRIPTION
**Pull request by nathanchance**
Switch to prebuilt toolchains wherever possible and use ccache.

All of the individual commits should be self explanatory but heckle me as necessary :)

### CODE REVIEW

**Review by msfjarvis — APPROVED** [review]
LGTM, great job putting this together :+1:

File: buildroot/arm.config [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 BR2_CCACHE=y
-BR2_OPTIMIZE_3=y
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TOOLCHAIN_EXTERNAL_BOOTLIN=y
 BR2_TOOLCHAIN_EXTERNAL_BOOTLIN_ARMV5_EABI_UCLIBC_STABLE=y
 BR2_TARGET_GENERIC_ROOT_PASSWD="root"
 BR2_TARGET_GENERIC_GETTY_PORT="ttyAMA0"
 BR2_ROOTFS_OVERLAY="../overlay-reboot"
-BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
-BR2_PACKAGE_BASH=y
```
Comment (nickdesaulniers) [inline buildroot/arm.config:9]: What shell do we boot to with `-i`? I assume we still have one, it's just whatever busybox has which I don't think is Bash?

Comment (nickdesaulniers) [reply buildroot/arm.config:9]: I use this with `-i` a lot, so I'm just checking this doesn't break that.

**Review by nickdesaulniers — APPROVED** [review]
Assuming this doesn't break `-i` workflows by removing bash.  Neat changes, did you find these as tips somewhere?

File: buildroot/arm.config (hunk 1 above)
Comment (tpimh) [reply buildroot/arm.config:9]: I believe the shell is busybox ash.

**Comment by nickdesaulniers** [comment]
I was thinking about this more overnight: does using ccache really speed up builds? I would expect to add to build time, unless you were to rebuild multiple images in repetition. Given how infrequently we update these, is it worth using ccache? Or does it somehow speed up the build regardless? I don't see how it would, unless some of the same code is being rebuilt for each image.  Since we're cross compiling, I'd expect many cache misses even for the same source inputs.

**Comment by nathanchance** [comment]
Yes, as Dima said, we have always been using busybox's ash except for on PowerPC it seems. Unless you were manually running `bash` after getting into the machine, there will be no visible difference.

Well, the prebuilt toolchains are infrequently updated so they should have some cache hits and the cache location is in `${HOME}` so it cannot be accidentally deleted but it seems that at least for me, there is no visible generation of a cache. I will delete it from the configurations and force push here shortly.

**Comment by nathanchance** [comment]
Alright, should be good, PTAL.

**Review by nickdesaulniers — APPROVED** [review]

**Pull request merged**